### ISSUE TITLE
CNIOBoringSSL compiler error on aarch64 Linux with Swift 5.1 and Swift 5.0.1

### ISSUE THREAD

**Issue by wlisac** [apple/swift-nio-ssl#140]
Hi there,

I'm attempting to build SwiftNIO SSL on aarch64 (ARMv8) Linux, but I'm running into a compiler error.

```plain
[199/450] Compiling CNIOBoringSSL crypto/fipsmodule/sha256-armv8.linux.aarch64.S
/swift-nio-ssl/Sources/CNIOBoringSSL/crypto/fipsmodule/sha256-armv8.linux.aarch64.S:76:10: error: expected compatible register or logical immediate
 tst w16,#ARMV8_SHA256
         ^
The command '/bin/sh -c swift build --jobs 1' returned a non-zero code: 1
```

This looks like it might be related to https://github.com/apple/swift-nio-ssl/issues/124.

### Steps to Reproduce
One way to reproduce this error is to compile SwiftNIO SSL using Docker Desktop for macOS with an aarch64 base image. Docker Desktop for macOS will use qemu to emulate the aarch64 architecture.

I maintain Docker images for Swift on ARM at [wlisac/swift-on-balena](https://github.com/wlisac/swift-on-balena) that we can use for the aarch64 base image.

**Dockerfile**

```Dockerfile
ARG FROM_IMAGE

FROM $FROM_IMAGE

RUN git clone https://github.com/apple/swift-nio-ssl.git

WORKDIR swift-nio-ssl

RUN swift package resolve

RUN swift build --jobs 1
```

**Build using Swift 5.1**
```bash
$ docker build --build-arg FROM_IMAGE=wlisac/aarch64-swift:5.1-build .
```

**Build using Swift 5.0.1**
```
$ docker build --build-arg FROM_IMAGE=wlisac/aarch64-swift:5.0.1-build .
```

Alternatively, you can reproduce by building SwiftNIO SSL on aarch64 hardware directly.

### Full Build Logs
**Swift 5.1**
```
docker build --build-arg FROM_IMAGE=wlisac/aarch64-swift:5.1-build .
Sending build context to Docker daemon  10.75kB
Step 1/6 : ARG FROM_IMAGE
Step 2/6 : FROM $FROM_IMAGE
 ---> fc67c19563ee
Step 3/6 : RUN git clone https://github.com/apple/swift-nio-ssl.git
 ---> Using cache
 ---> f92f27d89c07
Step 4/6 : WORKDIR swift-nio-ssl
 ---> Using cache
 ---> a984b72d8087
Step 5/6 : RUN swift package resolve
 ---> Using cache
 ---> 99b9325e7caa
Step 6/6 : RUN swift build --jobs 1
 ---> Running in 00d168bb8eed
[1/442] Compiling CNIOSHA1 c_nio_sha1.c
[2/442] Compiling CNIOLinux shim.c
[3/442] Compiling CNIOLinux ifaddrs-android.c
[4/442] Compiling CNIOHTTPParser c_nio_http_parser.c
[5/442] Compiling CNIODarwin shim.c
[6/442] Compiling CNIOBoringSSLShims shims.c
[7/442] Compiling sike.c
[8/442] Compiling isogeny.c
[9/442] Compiling fpx.c
[10/442] Compiling curve_params.c
[11/442] Compiling fp_generic.c
[12/442] Compiling p256.c
[13/442] Compiling curve25519.c
[14/442] Compiling tls_record.cc
[15/442] Compiling tls_method.cc
[16/442] Compiling tls13_server.cc
[17/442] Compiling tls13_enc.cc
[18/442] Compiling tls13_client.cc
[19/442] Compiling tls13_both.cc
[20/442] Compiling t1_lib.cc
[21/442] Compiling t1_enc.cc
[22/442] Compiling ssl_x509.cc
[23/442] Compiling ssl_versions.cc
[24/442] Compiling ssl_transcript.cc
[25/442] Compiling ssl_stat.cc
[26/442] Compiling ssl_session.cc
[27/442] Compiling ssl_privkey.cc
[28/442] Compiling ssl_lib.cc
[29/442] Compiling ssl_key_share.cc
[30/442] Compiling ssl_file.cc
[31/442] Compiling ssl_cipher.cc
[32/442] Compiling ssl_cert.cc
[33/442] Compiling ssl_buffer.cc
[34/442] Compiling ssl_asn1.cc
[35/442] Compiling ssl_aead_ctx.cc
[36/442] Compiling s3_pkt.cc
[37/442] Compiling s3_lib.cc
[38/442] Compiling s3_both.cc
[39/442] Compiling handshake_server.cc
[40/442] Compiling handshake_client.cc
[41/442] Compiling handshake.cc
[42/442] Compiling handoff.cc
[43/442] Compiling dtls_record.cc
[44/442] Compiling dtls_method.cc
[45/442] Compiling d1_srtp.cc
[46/442] Compiling d1_pkt.cc
[47/442] Compiling d1_lib.cc
[48/442] Compiling d1_both.cc
[49/442] Compiling bio_ssl.cc
[50/442] Compiling v3_utl.c
[51/442] Compiling v3_sxnet.c
[52/442] Compiling v3_skey.c
[53/442] Compiling v3_purp.c
[54/442] Compiling v3_prn.c
[55/442] Compiling v3_pmaps.c
[56/442] Compiling v3_pku.c
[57/442] Compiling v3_pcons.c
[58/442] Compiling v3_pcia.c
[59/442] Compiling v3_pci.c
[60/442] Compiling v3_ocsp.c
[61/442] Compiling v3_ncons.c
[62/442] Compiling v3_lib.c
[63/442] Compiling v3_int.c
[64/442] Compiling v3_info.c
[65/442] Compiling v3_ia5.c
[66/442] Compiling v3_genn.c
[67/442] Compiling v3_extku.c
[68/442] Compiling v3_enum.c
[69/442] Compiling v3_crld.c
[70/442] Compiling v3_cpols.c
[71/442] Compiling v3_conf.c
[72/442] Compiling v3_bitst.c
[73/442] Compiling v3_bcons.c
[74/442] Compiling v3_alt.c
[75/442] Compiling v3_akeya.c
[76/442] Compiling v3_akey.c
[77/442] Compiling pcy_tree.c
[78/442] Compiling pcy_node.c
[79/442] Compiling pcy_map.c
[80/442] Compiling pcy_lib.c
[81/442] Compiling pcy_data.c
[82/442] Compiling pcy_cache.c
[83/442] Compiling x_x509a.c
[84/442] Compiling x_x509.c
[85/442] Compiling x_val.c
[86/442] Compiling x_spki.c
[87/442] Compiling x_sig.c
[88/442] Compiling x_req.c
[89/442] Compiling x_pubkey.c
[90/442] Compiling x_pkey.c
[91/442] Compiling x_name.c
[92/442] Compiling x_info.c
[93/442] Compiling x_exten.c
[94/442] Compiling x_crl.c
[95/442] Compiling x_attrib.c
[96/442] Compiling x_all.c
[97/442] Compiling x_algor.c
[98/442] Compiling x509spki.c
[99/442] Compiling x509rset.c
[100/442] Compiling x509name.c
[101/442] Compiling x509cset.c
[102/442] Compiling x509_vpm.c
[103/442] Compiling x509_vfy.c
[104/442] Compiling x509_v3.c
[105/442] Compiling x509_txt.c
[106/442] Compiling x509_trs.c
[107/442] Compiling x509_set.c
[108/442] Compiling x509_req.c
[109/442] Compiling x509_r2x.c
[110/442] Compiling x509_obj.c
[111/442] Compiling x509_lu.c
[112/442] Compiling x509_ext.c
[113/442] Compiling x509_def.c
[114/442] Compiling x509_d2.c
[115/442] Compiling x509_cmp.c
[116/442] Compiling x509_att.c
[117/442] Compiling x509.c
[118/442] Compiling t_x509a.c
[119/442] Compiling t_x509.c
[120/442] Compiling t_req.c
[121/442] Compiling t_crl.c
[122/442] Compiling rsa_pss.c
[123/442] Compiling i2d_pr.c
[124/442] Compiling by_file.c
[125/442] Compiling by_dir.c
[126/442] Compiling asn1_gen.c
[127/442] Compiling algorithm.c
[128/442] Compiling a_verify.c
[129/442] Compiling a_strex.c
[130/442] Compiling a_sign.c
[131/442] Compiling a_digest.c
[132/442] Compiling thread_win.c
[133/442] Compiling thread_pthread.c
[134/442] Compiling thread_none.c
[135/442] Compiling thread.c
[136/442] Compiling fp-x86_64.mac.x86_64.S
[137/442] Compiling fp-x86_64.linux.x86_64.S
[138/442] Compiling fp-armv8.linux.aarch64.S
[139/442] Compiling fp-armv8.ios.aarch64.S
[140/442] Compiling stack.c
[141/442] Compiling siphash.c
[142/442] Compiling rsa_print.c
[143/442] Compiling rsa_asn1.c
[144/442] Compiling refcount_lock.c
[145/442] Compiling refcount_c11.c
[146/442] Compiling rc4.c
[147/442] Compiling windows.c
[148/442] Compiling rand_extra.c
[149/442] Compiling fuchsia.c
[150/442] Compiling forkunsafe.c
[151/442] Compiling deterministic.c
[152/442] Compiling pool.c
[153/442] Compiling poly1305_vec.c
[154/442] Compiling poly1305_arm_asm.S
[155/442] Compiling poly1305_arm.c
[156/442] Compiling poly1305.c
[157/442] Compiling pkcs8_x509.c
[158/442] Compiling pkcs8.c
[159/442] Compiling p5_pbev2.c
[160/442] Compiling pkcs7_x509.c
[161/442] Compiling pkcs7.c
[162/442] Compiling pem_xaux.c
[163/442] Compiling pem_x509.c
[164/442] Compiling pem_pkey.c
[165/442] Compiling pem_pk8.c
[166/442] Compiling pem_oth.c
[167/442] Compiling pem_lib.c
[168/442] Compiling pem_info.c
[169/442] Compiling pem_all.c
[170/442] Compiling obj_xref.c
[171/442] Compiling obj.c
[172/442] Compiling mem.c
[173/442] Compiling lhash.c
[174/442] Compiling hrss.c
[175/442] Compiling poly_rq_mul.S
[176/442] Compiling hkdf.c
[177/442] Compiling x86_64-mont5.mac.x86_64.S
[178/442] Compiling x86_64-mont5.linux.x86_64.S
[179/442] Compiling x86_64-mont.mac.x86_64.S
[180/442] Compiling x86_64-mont.linux.x86_64.S
[181/442] Compiling x86-mont.linux.x86.S
[182/442] Compiling vpaes-x86_64.mac.x86_64.S
[183/442] Compiling vpaes-x86_64.linux.x86_64.S
[184/442] Compiling vpaes-x86.linux.x86.S
[185/442] Compiling vpaes-armv8.linux.aarch64.S
[186/442] Compiling vpaes-armv8.ios.aarch64.S
[187/442] Compiling vpaes-armv7.linux.arm.S
[188/442] Compiling vpaes-armv7.ios.arm.S
[189/442] Compiling kdf.c
[190/442] Compiling sha512-x86_64.mac.x86_64.S
[191/442] Compiling sha512-x86_64.linux.x86_64.S
[192/442] Compiling sha512-armv8.linux.aarch64.S
[193/442] Compiling sha512-armv8.ios.aarch64.S
[194/442] Compiling sha512-armv4.linux.arm.S
[195/442] Compiling sha512-armv4.ios.arm.S
[196/442] Compiling sha512-586.linux.x86.S
[197/442] Compiling sha256-x86_64.mac.x86_64.S
[198/442] Compiling sha256-x86_64.linux.x86_64.S
/swift-nio-ssl/Sources/CNIOBoringSSL/crypto/fipsmodule/sha256-armv8.linux.aarch64.S:76:10: error: expected compatible register or logical immediate
 tst w16,#ARMV8_SHA256
         ^
[198/442] Compiling sha256-armv8.linux.aarch64.S
The command '/bin/sh -c swift build --jobs 1' returned a non-zero code: 1
```

**Swift 5.0.1**
```plain
docker build --build-arg FROM_IMAGE=wlisac/aarch64-swift:5.0.1-build .
Sending build context to Docker daemon  10.75kB
Step 1/6 : ARG FROM_IMAGE
Step 2/6 : FROM $FROM_IMAGE
5.0.1-build: Pulling from wlisac/aarch64-swift
85df13e07ac8: Pull complete 
8627bf1c6512: Pull complete 
c99ec04f4698: Pull complete 
dbc2e75663b5: Pull complete 
ea41997c9a0a: Pull complete 
9427b9020797: Pull complete 
69a330048877: Pull complete 
9716a833dab9: Pull complete 
f06d72f23eca: Pull complete 
c114c6652bee: Pull complete 
4bb85fafe2f6: Pull complete 
Digest: sha256:d69a0d81de55f32d5392b888bfd95c7dbdb04357a45a1c33b56802b9ddc4e56d
Status: Downloaded newer image for wlisac/aarch64-swift:5.0.1-build
 ---> b79bef9417ba
Step 3/6 : RUN git clone https://github.com/apple/swift-nio-ssl.git
 ---> Running in 11f033301f3d
Cloning into 'swift-nio-ssl'...
Removing intermediate container 11f033301f3d
 ---> 140795c0b56f
Step 4/6 : WORKDIR swift-nio-ssl
 ---> Running in efd5b64d941e
Removing intermediate container efd5b64d941e
 ---> f685d3db6142
Step 5/6 : RUN swift package resolve
 ---> Running in 0f3c0fa5023a
Fetching https://github.com/apple/swift-nio.git
Completed resolution in 22.90s
Cloning https://github.com/apple/swift-nio.git
Resolving https://github.com/apple/swift-nio.git at 2.8.0
Removing intermediate container 0f3c0fa5023a
 ---> 956b0b9fe0a1
Step 6/6 : RUN swift build --jobs 1
 ---> Running in 4e6721195e5a
[1/450] Compiling CNIOSHA1 c_nio_sha1.c
[2/450] Compiling CNIOLinux shim.c
[3/450] Compiling CNIOLinux ifaddrs-android.c
[4/450] Compiling CNIOHTTPParser c_nio_http_parser.c
[5/450] Compiling CNIODarwin shim.c
[6/450] Compiling CNIOBoringSSLShims shims.c
[7/450] Compiling CNIOBoringSSL third_party/sike/sike.c
[8/450] Compiling CNIOBoringSSL third_party/sike/isogeny.c
[9/450] Compiling CNIOBoringSSL third_party/sike/fpx.c
[10/450] Compiling CNIOBoringSSL third_party/sike/curve_params.c
[11/450] Compiling CNIOBoringSSL third_party/sike/asm/fp_generic.c
[12/450] Compiling CNIOBoringSSL third_party/fiat/p256.c
[13/450] Compiling CNIOBoringSSL third_party/fiat/curve25519.c
[14/450] Compiling CNIOBoringSSL ssl/tls_record.cc
[15/450] Compiling CNIOBoringSSL ssl/tls_method.cc
[16/450] Compiling CNIOBoringSSL ssl/tls13_server.cc
[17/450] Compiling CNIOBoringSSL ssl/tls13_enc.cc
[18/450] Compiling CNIOBoringSSL ssl/tls13_client.cc
[19/450] Compiling CNIOBoringSSL ssl/tls13_both.cc
[20/450] Compiling CNIOBoringSSL ssl/t1_lib.cc
[21/450] Compiling CNIOBoringSSL ssl/t1_enc.cc
[22/450] Compiling CNIOBoringSSL ssl/ssl_x509.cc
[23/450] Compiling CNIOBoringSSL ssl/ssl_versions.cc
[24/450] Compiling CNIOBoringSSL ssl/ssl_transcript.cc
[25/450] Compiling CNIOBoringSSL ssl/ssl_stat.cc
[26/450] Compiling CNIOBoringSSL ssl/ssl_session.cc
[27/450] Compiling CNIOBoringSSL ssl/ssl_privkey.cc
[28/450] Compiling CNIOBoringSSL ssl/ssl_lib.cc
[29/450] Compiling CNIOBoringSSL ssl/ssl_key_share.cc
[30/450] Compiling CNIOBoringSSL ssl/ssl_file.cc
[31/450] Compiling CNIOBoringSSL ssl/ssl_cipher.cc
[32/450] Compiling CNIOBoringSSL ssl/ssl_cert.cc
[33/450] Compiling CNIOBoringSSL ssl/ssl_buffer.cc
[34/450] Compiling CNIOBoringSSL ssl/ssl_asn1.cc
[35/450] Compiling CNIOBoringSSL ssl/ssl_aead_ctx.cc
[36/450] Compiling CNIOBoringSSL ssl/s3_pkt.cc
[37/450] Compiling CNIOBoringSSL ssl/s3_lib.cc
[38/450] Compiling CNIOBoringSSL ssl/s3_both.cc
[39/450] Compiling CNIOBoringSSL ssl/handshake_server.cc
[40/450] Compiling CNIOBoringSSL ssl/handshake_client.cc
[41/450] Compiling CNIOBoringSSL ssl/handshake.cc
[42/450] Compiling CNIOBoringSSL ssl/handoff.cc
[43/450] Compiling CNIOBoringSSL ssl/dtls_record.cc
[44/450] Compiling CNIOBoringSSL ssl/dtls_method.cc
[45/450] Compiling CNIOBoringSSL ssl/d1_srtp.cc
[46/450] Compiling CNIOBoringSSL ssl/d1_pkt.cc
[47/450] Compiling CNIOBoringSSL ssl/d1_lib.cc
[48/450] Compiling CNIOBoringSSL ssl/d1_both.cc
[49/450] Compiling CNIOBoringSSL ssl/bio_ssl.cc
[50/450] Compiling CNIOBoringSSL crypto/x509v3/v3_utl.c
[51/450] Compiling CNIOBoringSSL crypto/x509v3/v3_sxnet.c
[52/450] Compiling CNIOBoringSSL crypto/x509v3/v3_skey.c
[53/450] Compiling CNIOBoringSSL crypto/x509v3/v3_purp.c
[54/450] Compiling CNIOBoringSSL crypto/x509v3/v3_prn.c
[55/450] Compiling CNIOBoringSSL crypto/x509v3/v3_pmaps.c
[56/450] Compiling CNIOBoringSSL crypto/x509v3/v3_pku.c
[57/450] Compiling CNIOBoringSSL crypto/x509v3/v3_pcons.c
[58/450] Compiling CNIOBoringSSL crypto/x509v3/v3_pcia.c
[59/450] Compiling CNIOBoringSSL crypto/x509v3/v3_pci.c
[60/450] Compiling CNIOBoringSSL crypto/x509v3/v3_ocsp.c
[61/450] Compiling CNIOBoringSSL crypto/x509v3/v3_ncons.c
[62/450] Compiling CNIOBoringSSL crypto/x509v3/v3_lib.c
[63/450] Compiling CNIOBoringSSL crypto/x509v3/v3_int.c
[64/450] Compiling CNIOBoringSSL crypto/x509v3/v3_info.c
[65/450] Compiling CNIOBoringSSL crypto/x509v3/v3_ia5.c
[66/450] Compiling CNIOBoringSSL crypto/x509v3/v3_genn.c
[67/450] Compiling CNIOBoringSSL crypto/x509v3/v3_extku.c
[68/450] Compiling CNIOBoringSSL crypto/x509v3/v3_enum.c
[69/450] Compiling CNIOBoringSSL crypto/x509v3/v3_crld.c
[70/450] Compiling CNIOBoringSSL crypto/x509v3/v3_cpols.c
[71/450] Compiling CNIOBoringSSL crypto/x509v3/v3_conf.c
[72/450] Compiling CNIOBoringSSL crypto/x509v3/v3_bitst.c
[73/450] Compiling CNIOBoringSSL crypto/x509v3/v3_bcons.c
[74/450] Compiling CNIOBoringSSL crypto/x509v3/v3_alt.c
[75/450] Compiling CNIOBoringSSL crypto/x509v3/v3_akeya.c
[76/450] Compiling CNIOBoringSSL crypto/x509v3/v3_akey.c
[77/450] Compiling CNIOBoringSSL crypto/x509v3/pcy_tree.c
[78/450] Compiling CNIOBoringSSL crypto/x509v3/pcy_node.c
[79/450] Compiling CNIOBoringSSL crypto/x509v3/pcy_map.c
[80/450] Compiling CNIOBoringSSL crypto/x509v3/pcy_lib.c
[81/450] Compiling CNIOBoringSSL crypto/x509v3/pcy_data.c
[82/450] Compiling CNIOBoringSSL crypto/x509v3/pcy_cache.c
[83/450] Compiling CNIOBoringSSL crypto/x509/x_x509a.c
[84/450] Compiling CNIOBoringSSL crypto/x509/x_x509.c
[85/450] Compiling CNIOBoringSSL crypto/x509/x_val.c
[86/450] Compiling CNIOBoringSSL crypto/x509/x_spki.c
[87/450] Compiling CNIOBoringSSL crypto/x509/x_sig.c
[88/450] Compiling CNIOBoringSSL crypto/x509/x_req.c
[89/450] Compiling CNIOBoringSSL crypto/x509/x_pubkey.c
[90/450] Compiling CNIOBoringSSL crypto/x509/x_pkey.c
[91/450] Compiling CNIOBoringSSL crypto/x509/x_name.c
[92/450] Compiling CNIOBoringSSL crypto/x509/x_info.c
[93/450] Compiling CNIOBoringSSL crypto/x509/x_exten.c
[94/450] Compiling CNIOBoringSSL crypto/x509/x_crl.c
[95/450] Compiling CNIOBoringSSL crypto/x509/x_attrib.c
[96/450] Compiling CNIOBoringSSL crypto/x509/x_all.c
[97/450] Compiling CNIOBoringSSL crypto/x509/x_algor.c
[98/450] Compiling CNIOBoringSSL crypto/x509/x509spki.c
[99/450] Compiling CNIOBoringSSL crypto/x509/x509rset.c
[100/450] Compiling CNIOBoringSSL crypto/x509/x509name.c
[101/450] Compiling CNIOBoringSSL crypto/x509/x509cset.c
[102/450] Compiling CNIOBoringSSL crypto/x509/x509_vpm.c
[103/450] Compiling CNIOBoringSSL crypto/x509/x509_vfy.c
[104/450] Compiling CNIOBoringSSL crypto/x509/x509_v3.c
[105/450] Compiling CNIOBoringSSL crypto/x509/x509_txt.c
[106/450] Compiling CNIOBoringSSL crypto/x509/x509_trs.c
[107/450] Compiling CNIOBoringSSL crypto/x509/x509_set.c
[108/450] Compiling CNIOBoringSSL crypto/x509/x509_req.c
[109/450] Compiling CNIOBoringSSL crypto/x509/x509_r2x.c
[110/450] Compiling CNIOBoringSSL crypto/x509/x509_obj.c
[111/450] Compiling CNIOBoringSSL crypto/x509/x509_lu.c
[112/450] Compiling CNIOBoringSSL crypto/x509/x509_ext.c
[113/450] Compiling CNIOBoringSSL crypto/x509/x509_def.c
[114/450] Compiling CNIOBoringSSL crypto/x509/x509_d2.c
[115/450] Compiling CNIOBoringSSL crypto/x509/x509_cmp.c
[116/450] Compiling CNIOBoringSSL crypto/x509/x509_att.c
[117/450] Compiling CNIOBoringSSL crypto/x509/x509.c
[118/450] Compiling CNIOBoringSSL crypto/x509/t_x509a.c
[119/450] Compiling CNIOBoringSSL crypto/x509/t_x509.c
[120/450] Compiling CNIOBoringSSL crypto/x509/t_req.c
[121/450] Compiling CNIOBoringSSL crypto/x509/t_crl.c
[122/450] Compiling CNIOBoringSSL crypto/x509/rsa_pss.c
[123/450] Compiling CNIOBoringSSL crypto/x509/i2d_pr.c
[124/450] Compiling CNIOBoringSSL crypto/x509/by_file.c
[125/450] Compiling CNIOBoringSSL crypto/x509/by_dir.c
[126/450] Compiling CNIOBoringSSL crypto/x509/asn1_gen.c
[127/450] Compiling CNIOBoringSSL crypto/x509/algorithm.c
[128/450] Compiling CNIOBoringSSL crypto/x509/a_verify.c
[129/450] Compiling CNIOBoringSSL crypto/x509/a_strex.c
[130/450] Compiling CNIOBoringSSL crypto/x509/a_sign.c
[131/450] Compiling CNIOBoringSSL crypto/x509/a_digest.c
[132/450] Compiling CNIOBoringSSL crypto/thread_win.c
[133/450] Compiling CNIOBoringSSL crypto/thread_pthread.c
[134/450] Compiling CNIOBoringSSL crypto/thread_none.c
[135/450] Compiling CNIOBoringSSL crypto/thread.c
[136/450] Compiling CNIOBoringSSL crypto/third_party/sike/asm/fp-x86_64.mac.x86_64.S
[137/450] Compiling CNIOBoringSSL crypto/third_party/sike/asm/fp-x86_64.linux.x86_64.S
[138/450] Compiling CNIOBoringSSL crypto/third_party/sike/asm/fp-armv8.linux.aarch64.S
[139/450] Compiling CNIOBoringSSL crypto/third_party/sike/asm/fp-armv8.ios.aarch64.S
[140/450] Compiling CNIOBoringSSL crypto/stack/stack.c
[141/450] Compiling CNIOBoringSSL crypto/siphash/siphash.c
[142/450] Compiling CNIOBoringSSL crypto/rsa_extra/rsa_print.c
[143/450] Compiling CNIOBoringSSL crypto/rsa_extra/rsa_asn1.c
[144/450] Compiling CNIOBoringSSL crypto/refcount_lock.c
[145/450] Compiling CNIOBoringSSL crypto/refcount_c11.c
[146/450] Compiling CNIOBoringSSL crypto/rc4/rc4.c
[147/450] Compiling CNIOBoringSSL crypto/rand_extra/windows.c
[148/450] Compiling CNIOBoringSSL crypto/rand_extra/rand_extra.c
[149/450] Compiling CNIOBoringSSL crypto/rand_extra/fuchsia.c
[150/450] Compiling CNIOBoringSSL crypto/rand_extra/forkunsafe.c
[151/450] Compiling CNIOBoringSSL crypto/rand_extra/deterministic.c
[152/450] Compiling CNIOBoringSSL crypto/pool/pool.c
[153/450] Compiling CNIOBoringSSL crypto/poly1305/poly1305_vec.c
[154/450] Compiling CNIOBoringSSL crypto/poly1305/poly1305_arm_asm.S
[155/450] Compiling CNIOBoringSSL crypto/poly1305/poly1305_arm.c
[156/450] Compiling CNIOBoringSSL crypto/poly1305/poly1305.c
[157/450] Compiling CNIOBoringSSL crypto/pkcs8/pkcs8_x509.c
[158/450] Compiling CNIOBoringSSL crypto/pkcs8/pkcs8.c
[159/450] Compiling CNIOBoringSSL crypto/pkcs8/p5_pbev2.c
[160/450] Compiling CNIOBoringSSL crypto/pkcs7/pkcs7_x509.c
[161/450] Compiling CNIOBoringSSL crypto/pkcs7/pkcs7.c
[162/450] Compiling CNIOBoringSSL crypto/pem/pem_xaux.c
[163/450] Compiling CNIOBoringSSL crypto/pem/pem_x509.c
[164/450] Compiling CNIOBoringSSL crypto/pem/pem_pkey.c
[165/450] Compiling CNIOBoringSSL crypto/pem/pem_pk8.c
[166/450] Compiling CNIOBoringSSL crypto/pem/pem_oth.c
[167/450] Compiling CNIOBoringSSL crypto/pem/pem_lib.c
[168/450] Compiling CNIOBoringSSL crypto/pem/pem_info.c
[169/450] Compiling CNIOBoringSSL crypto/pem/pem_all.c
[170/450] Compiling CNIOBoringSSL crypto/obj/obj_xref.c
[171/450] Compiling CNIOBoringSSL crypto/obj/obj.c
[172/450] Compiling CNIOBoringSSL crypto/mem.c
[173/450] Compiling CNIOBoringSSL crypto/lhash/lhash.c
[174/450] Compiling CNIOBoringSSL crypto/hrss/hrss.c
[175/450] Compiling CNIOBoringSSL crypto/hrss/asm/poly_rq_mul.S
[176/450] Compiling CNIOBoringSSL crypto/hkdf/hkdf.c
[177/450] Compiling CNIOBoringSSL crypto/fipsmodule/x86_64-mont5.mac.x86_64.S
[178/450] Compiling CNIOBoringSSL crypto/fipsmodule/x86_64-mont5.linux.x86_64.S
[179/450] Compiling CNIOBoringSSL crypto/fipsmodule/x86_64-mont.mac.x86_64.S
[180/450] Compiling CNIOBoringSSL crypto/fipsmodule/x86_64-mont.linux.x86_64.S
[181/450] Compiling CNIOBoringSSL crypto/fipsmodule/x86-mont.linux.x86.S
[182/450] Compiling CNIOBoringSSL crypto/fipsmodule/vpaes-x86_64.mac.x86_64.S
[183/450] Compiling CNIOBoringSSL crypto/fipsmodule/vpaes-x86_64.linux.x86_64.S
[184/450] Compiling CNIOBoringSSL crypto/fipsmodule/vpaes-x86.linux.x86.S
[185/450] Compiling CNIOBoringSSL crypto/fipsmodule/vpaes-armv8.linux.aarch64.S
[186/450] Compiling CNIOBoringSSL crypto/fipsmodule/vpaes-armv8.ios.aarch64.S
[187/450] Compiling CNIOBoringSSL crypto/fipsmodule/vpaes-armv7.linux.arm.S
[188/450] Compiling CNIOBoringSSL crypto/fipsmodule/vpaes-armv7.ios.arm.S
[189/450] Compiling CNIOBoringSSL crypto/fipsmodule/tls/kdf.c
[190/450] Compiling CNIOBoringSSL crypto/fipsmodule/sha512-x86_64.mac.x86_64.S
[191/450] Compiling CNIOBoringSSL crypto/fipsmodule/sha512-x86_64.linux.x86_64.S
[192/450] Compiling CNIOBoringSSL crypto/fipsmodule/sha512-armv8.linux.aarch64.S
[193/450] Compiling CNIOBoringSSL crypto/fipsmodule/sha512-armv8.ios.aarch64.S
[194/450] Compiling CNIOBoringSSL crypto/fipsmodule/sha512-armv4.linux.arm.S
[195/450] Compiling CNIOBoringSSL crypto/fipsmodule/sha512-armv4.ios.arm.S
[196/450] Compiling CNIOBoringSSL crypto/fipsmodule/sha512-586.linux.x86.S
[197/450] Compiling CNIOBoringSSL crypto/fipsmodule/sha256-x86_64.mac.x86_64.S
[198/450] Compiling CNIOBoringSSL crypto/fipsmodule/sha256-x86_64.linux.x86_64.S
[199/450] Compiling CNIOBoringSSL crypto/fipsmodule/sha256-armv8.linux.aarch64.S
/swift-nio-ssl/Sources/CNIOBoringSSL/crypto/fipsmodule/sha256-armv8.linux.aarch64.S:76:10: error: expected compatible register or logical immediate
 tst w16,#ARMV8_SHA256
         ^
The command '/bin/sh -c swift build --jobs 1' returned a non-zero code: 1
```

**Comment by Lukasa** [comment]
Yup, agree that this seems related to #124. Let's dig in and see what we can see.

**Comment by Lukasa** [comment]
Looks like the fix needs to be the same as #125, but adding `__aarch64__` to `scripts/patch-2-arm-arch.patch` in addition to the already existing `__arm64__` and `__arm__`. Are you open to making that patch @wlisac?

**Comment by wlisac** [comment]
@Lukasa PR opened 🙂

Thanks for confirming the issue / fix.

**Comment by Lukasa** [comment]
Fixed in 2.4.2, thanks!
